tinytga/src/lib.rs

```diff
@@ -371,7 +386,19 @@ impl<'a> Iterator for TgaIterator<'a> {
 
         if self.x >= u32::from(self.tga.width()) {
             self.x = 0;
-            self.y += 1;
+
+            if self.tga.header.image_origin.is_bottom() {
+                if self.y > 0 {
+                    self.y -= 1;
+                } else {
+                    self.done = true;
+                }
+            } else {
+                self.y += 1;
+                if self.y >= u32::from(self.tga.header.height) {
```

Comment:
We do quite a lot of `u32::from` in this function and elsewhere. Would it be better to use `u16`s in the iterator and maybe also `Pixel`? This would make the tinytga semantically correct as TGA images can't be larger than a `u16`'s max value in either dimension anyway. The type coercion would be pushed the e-g portion of the crate, but we already do a `as i32` for pixel coordinates there so I don't think that would need any changes.

Reply:
Sounds like a good idea, but isn't directly related to this PR IMO. Or do you want me to include the change in this PR?

Reply:
It's slightly outside the original scope, I agree. If you want to keep it separate can you open a tracking issue please? It might also be worth checking `tinybmp` and doing the same there too.

Reply:
I've changed my mind and included the change in this PR.